legacy/builder/phases/libraries_builder.go

```diff
@@ -134,8 +134,14 @@ func fixLDFLAG(ctx *types.Context, libs libraries.List) error {
 			}
 		}
 
-		currLDFlags := ctx.BuildProperties.Get(constants.BUILD_PROPERTIES_COMPILER_LIBRARIES_LDFLAGS)
-		ctx.BuildProperties.Set(constants.BUILD_PROPERTIES_COMPILER_LIBRARIES_LDFLAGS, currLDFlags+"\"-L"+path.String()+"\" "+libsCmd+" ")
+		key := "compiler.libraries.ldflags"
```

Comment:
Since the linking order is crucial here, I would avoid the fallback if it **can** fail

Reply:
I've removed the extra commit, it's worth keeping the PR clean anyway IMHO